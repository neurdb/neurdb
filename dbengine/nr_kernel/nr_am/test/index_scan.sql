DROP TABLE IF EXISTS xs CASCADE;
CREATE TABLE xs (
                    id  INT,
                    tag TEXT,
                    val TEXT,
                    PRIMARY KEY (id)
) USING nram;

CREATE INDEX xs_tag_idx ON xs(tag);

-- Insert dummy data.
INSERT INTO xs
SELECT i, 'tag' || (i % 10), 'val' || i
FROM generate_series(1, 50) AS i;

SET enable_seqscan = OFF;

-- Test equality scan on primary key.
EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF)
SELECT * FROM xs WHERE id = 42;

SELECT * FROM xs WHERE id = 42;

-- Test equality scan on secondary index.
EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF)
SELECT * FROM xs WHERE tag = 'tag3';

SELECT * FROM xs WHERE tag = 'tag3';

-- Test range scan on primary key.
EXPLAIN (ANALYZE, VERBOSE, COSTS OFF, TIMING OFF)
SELECT * FROM xs WHERE id > 45;

SELECT * FROM xs WHERE id > 45;

-- Cleanup.
DROP TABLE xs;
SET enable_seqscan = ON;
