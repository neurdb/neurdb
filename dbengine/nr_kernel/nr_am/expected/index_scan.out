DROP TABLE IF EXISTS xs CASCADE;
NOTICE:  table "xs" does not exist, skipping
CREATE TABLE xs (
                    id  INT,
                    tag TEXT,
                    val TEXT,
                    PRIMARY KEY (id)
) USING nram;
CREATE INDEX xs_tag_idx ON xs(tag);
-- Insert dummy data.
INSERT INTO xs
SELECT i, 'tag' || (i % 10), 'val' || i
FROM generate_series(1, 50) AS i;
SET enable_seqscan = OFF;
-- Test equality scan on primary key.
SELECT * FROM xs WHERE id = 42;
 id | tag  |  val
----+------+-------
 42 | tag2 | val42
(1 row)

-- Test equality scan on secondary index.
SELECT * FROM xs WHERE tag = 'tag3';
 id | tag  |  val
----+------+-------
  3 | tag3 | val3
 13 | tag3 | val13
 23 | tag3 | val23
 33 | tag3 | val33
 43 | tag3 | val43
(5 rows)

-- Test range scan on primary key.
SELECT * FROM xs WHERE id > 45;
 id | tag  |  val
----+------+-------
 46 | tag6 | val46
 47 | tag7 | val47
 48 | tag8 | val48
 49 | tag9 | val49
 50 | tag0 | val50
(5 rows)

-- Cleanup.
DROP TABLE xs;
SET enable_seqscan = ON;
